models/multiphase/d3q27_pf_velocity/Boundary.c.Rt

```diff
@@ -70,42 +70,131 @@ CudaDeviceFunction void <?R C(my_velocity_boundaries[ii])?>(){
 		C(h[sel], heq[sel] + heq[bounce][sel] - h[bounce][sel] - 0.5 * Unknowns * (exM))
 	?>
 }
+<?R
+}
+?>
 
-CudaDeviceFunction void <?R C(my_pressure_boundaries[ii])?>(){
+<?R
+for (ii in 1:cube_faces){
+for (j in 1:length(pressure_boundary_types)) {  ?>
+CudaDeviceFunction void <?%s paste0(C(my_pressure_boundaries[ii]), pressure_boundary_types[j]) ?>(){
     real_t d = getRho();
-	real_t pstar = Pressure / (d*cs2);
-	<?R
-        EQ = MRT_eq(U,PV(1),u,mat=t(M))
-       	pstar = PV("pstar")
-		n = my_normal_directions[ii,]
-		geq = pstar*w_g + (EQ$feq - w_g)
-		bounce = Bounce(U)
-		sel = as.vector( (U %*% n) < 0)
+    real_t myPhaseField = 0.0;
 
-		sel2 = as.vector( ( U %*% n) == 0)	
-		exM = (g[sel2] - geq[sel2]) %*% rep(1,sum(sel2)) 
-		Unknowns = 1.0/sum(sel)
+    if ((NodeType & NODE_BOUNDARY) == <?%s paste0("NODE_WPressure", pressure_boundary_types[j]) ?>){
+        d = InvasionDrainage == 2? Density_l : Density_h;
+        myPhaseField = InvasionDrainage == 2? 0.0 : 1.0;
+    }
 
-		C(g[sel], geq[sel] + geq[bounce][sel] - g[bounce][sel] - 0.5 * Unknowns * (exM))
-	?>
-	<?R
-		U_PF = U[1:PF_velocities,]
-       	pf = PV("PhaseField")
-		n = my_normal_directions[ii,]
-		heq = pf * EQ_h$feq
-		bounce = Bounce(U_PF)
-		sel = as.vector( (U_PF %*% n) < 0)
+    if ((NodeType & NODE_BOUNDARY) == <?%s paste0("NODE_EPressure", pressure_boundary_types[j]) ?>){
+       d = InvasionDrainage == 2? Density_h : Density_l;
+       myPhaseField = InvasionDrainage == 2? 1.0 : 0.0;
+    }
 
```

Comment:
@shkodm I think here you should be setting:
`
PhaseF = PhaseField
d = getRho()
`
I'm not sure what your InvasionDrainage setting here is checking as you specify it should be 1 or -1 in the Dynamics.R. This is causing an issue (i think) for pressure pressure cases where fpf is used as it effectively forces PhaseField=0 on all pressure boundaries.

Reply:
@TravisMitchell The documentation in `Dynamics.R` is incorrect (copypasted), `InvasionDrainage` should have values 1 or 2. `d` should be set explicitly because I don't want any streamed values obtained from `getRho()` This was the whole point of those if statements, without it would not work properly.

Reply:
@shkodm, wouldn't it be much simpler/neater to take `PhaseField` (zonal setting defined in xml) and then interpolate for `d` based off `Density_h` and `Density_l`. This would mean that the `myPhaseField` would be simply set from the zonal condition applied in the xml rather than relying on this `InvasionDrainage` setting.

Reply:
@TravisMitchell yea, interpolation is a good idea. I think I implemented it with an additional setting originally, because `InvasionDrainage` (or whatever the name of the flag) is still necessary to  know if we want to "reflect/mirror" the boundaries on the inlet (to have proper wall normal not pointing to the other side of domain), or if we want to keep periodic boundaries. 

Anyway, I will remove the if statements here later today or tomorrow and double check the boundary conditions.